src/components/tables/Details/StatusIndicatorAndLabel.tsx

```diff
@@ -1,35 +1,45 @@
 import { TableCell, Typography, useTheme } from '@mui/material';
+import { ShardDetailStoreNames, useZustandStore } from 'context/Zustand';
 import { Shard } from 'data-plane-gateway/types/shard_client';
-import { useRouteStore } from 'hooks/useRouteStore';
 import { FormattedMessage } from 'react-intl';
-import { shardDetailSelectors, ShardStatusColor } from 'stores/ShardDetail';
+import {
+    shardDetailSelectors,
+    ShardDetailStore,
+    ShardStatusColor,
+} from 'stores/ShardDetail';
 
 interface Props {
     shard: Shard;
+    shardDetailStoreName: ShardDetailStoreNames;
 }
 
-function StatusIndicatorAndLabel({ shard }: Props) {
+function StatusIndicatorAndLabel({ shard, shardDetailStoreName }: Props) {
     const { id } = shard.spec;
+    const shardId = id ?? '';
```

Comment:
Every shard should have an ID. However, most of the interfaces supplied by the data plane gateway have optional properties because of the mechanism used to type the assets in that project.